error.go

```diff
@@ -194,3 +200,53 @@ func wrapIfContextError(err error) error {
 	}
 	return err
 }
+
+// wrapIfLikelyWithGRPCNotUsedError adds a wrapping error that has a message
+// telling the caller that they likely need to use h2c but are using a raw http.Client{}.
+//
+// This happens when running a gRPC-only server.
+// This is fragile and may break over time, and this should be considered a best-effort.
+func wrapIfLikelyH2CNotConfiguredError(request *http.Request, err error) error {
+	if err == nil {
+		return nil
+	}
+	if _, ok := asError(err); ok {
+		return err
+	}
+	if url := request.URL; url != nil && url.Scheme == "https" {
```

Comment:
Can we bail out if the scheme is anything other than "http"? As written, we're still returning our special h2c error strings for file:// and unix:// URLs.

Reply:
Done